### ISSUE TITLE
Support spy-like support for previously enqueued jobs

### ISSUE THREAD

**Issue by agrobbin** [gocardless/rspec-activejob#4]
With [rspec-sidekiq](https://github.com/philostler/rspec-sidekiq) we have been able to assert that we enqueued a job (or multiple jobs) after running the code that actually enqueues the job. Example:

```ruby
describe "#requery!" do
  subject(:record) { Record.new(id: 123) }

  before { record.requery! }

  it "performs requerying in the background" do
    expect(RequeryJob).to have_enqueued_job(123)
  end
end
```

In this simple case, your `enqueue_a` matcher will work just fine. However, if there are several other `it` blocks that are asserting other pieces of the `requery!` method, then the spec becomes more repetitive if we have to call `record.requery!` within each `it` block rather than in the `before`.

It'd be great if there was an equivalent spy-like matcher within `rspec-activejob`! Not sure if you're looking to start accepting PRs for this, as it's only a week old, so thought I'd put out the issue first.

**Comment by isaacseymour** [comment]
Hey, sorry I forgot to watch my own repo so didn't see this until now - PRs are very much welcome! That seems like a pretty sensible feature - I'll have a go at implementing it (probably `expect(JobClass).to have_been_enqueued.with(...))`) when I have some time, unless you beat me to it :)

**Comment by aprescott** [comment]
+1 on this matcher!

I'm not sure about the naming of `have_been_enqueued` though. To me, `have_enqueued_job` is about verifying that `FooJob` should "have (an) enqueued job (with the arguments) ____". `FooJob` itself being enqueued is a bit of an awkward naming.

**Comment by isaacseymour** [comment]
Hmm, I dislike `have_enqueued_job` because it sounds a bit like `FooJob` did the enqueuing itself (which it didn't - ActiveJob did that _to_ `FooJob`). Would also fit better with `expect { thing }.to enqueue_a(JobClass)` - the block does the enqueuing, and the `JobClass` gets enqueued. I've implemented this in #7 and will probably merge it tomorrow unless I'm persuaded otherwise.
